Syncthing panics with 'interface conversion: interface ...'

I've noticed that Syncthing 0.14.51 crashes on my Arch Linux system with the following error:   
```
panic: interface conversion: interface {} is map[string]string, not map[string]interface {}
```
Full error log: http://paste.opensuse.org/96753389

Here's how I could reproduce the issue:
1. Copy https://github.com/Martchus/syncthingtray/blob/master/connector/testfiles/testconfig/config.xml to an empty directory
2. Start `syncthing -gui-address=http://localhost:4001 -gui-apikey=syncthingtestinstance -home=/dir/with/config -no-browser -verbose`

Further notes:
* The problem only occurs if device `MMGUI6U-WUEZQCP-XZZ6VYB-LCT4TVC-ER2HAVX-QYT6X7D-S6ZSG2B-323KLQ7` is actually reachable in my local network.
* This device in my local network has actually a different device ID than stated in the configuration file. That 'misconfiguration' comes from the fact that https://github.com/Martchus/syncthingtray/blob/master/connector/testfiles/testconfig/config.xml is really just intended for testing Syncthing Tray's code and no actual connection is supposed to happen.
* I have another Syncthing instance running at the same time which is configured correctly and it can connect to the mentioned device in my local network.

Comments from @imsodin via IRC:
> [23:50] \<imsodin\> Martchus: The log is pretty clear, looks like a straight forward bug. what's not straight forward is, that it is present since an eternity and you seem to be the first to observe it!?
> [23:50] \<imsodin\> could you file it on github please
> [00:06] \<Martchus\> to me 'panic: interface conversion: interface {} is map[string]string, not map[string]interface {}' mean litterally nothing - likely some (de)serialization code - but it doesn't point to a file or line of code
> [00:13] \<imsodin\> the next block contains the info about which line in the code this originates.
> [00:13] \<imsodin\> then I grep for "DeviceRejected" and see that the type in which it is logged is different from the type in the assertion -> boom
> [00:14] \<imsodin\> i can reliably reproduce by starting a fresh instance with -verbose and by adding this device on a remote device. as soon as the "invitation" reaches here, it panics
> [00:15] \<imsodin\> i suspect this did not surface due to suture handling the panics before

--- a/cmd/syncthing/verboseservice.go
+++ b/cmd/syncthing/verboseservice.go
@@ -105,7 +105,7 @@ func (s *verboseService) formatEvent(ev events.Event) string {
 		return fmt.Sprintf("Device %v sent an index update for %q with %d items", data["device"], data["folder"], data["items"])
 
 	case events.DeviceRejected:
-		data := ev.Data.(map[string]interface{})
+		data := ev.Data.(map[string]string)
 		return fmt.Sprintf("Rejected connection from device %v at %v", data["device"], data["address"])
 
 	case events.FolderRejected:

